feat: add `stats/incr/incrnanmae`

Adithya-jh opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves #5563 .

## Description

> What is the purpose of this pull request?

This pull request:

-  Adds the @stdlib/stats/incr/nanmae package, which computes an incremental mean absolute error while handling NaN values by skipping them.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5563 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

Benchmark:
<img width="891" alt="Screenshot 2025-03-15 at 12 51 38 PM" src="https://github.com/user-attachments/assets/4fba1e31-05a5-4618-8e87-6cceb5b0e74a" />

Tests:
<img width="891" alt="Screenshot 2025-03-15 at 12 53 29 PM" src="https://github.com/user-attachments/assets/5f6b249f-6d5b-4dad-9007-c9c377f7948a" />

Example:
<img width="891" alt="Screenshot 2025-03-15 at 12 54 30 PM" src="https://github.com/user-attachments/assets/2ad88766-4199-4a22-aa0f-523709bbf97d" />



## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Adithya-jh committed:
feat(stats/incr): add incrnanmae accumulator

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

Adithya-jh commented:
Hi @kgryte , i have added the incrnanmae accumulator, could u pls verify !
Thank you !

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmae/package.json:
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "@stdlib/stats/incr/nanmae",
+  "version": "0.0.0",
+  "description": "Compute the mean absolute error incrementally, ignoring NaN values..",
```
Double period.

```suggestion
  "description": "Compute the mean absolute error incrementally, ignoring NaN values.",
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmae/examples/index.js:
```diff
@@ -0,0 +1,41 @@
+	v1 = ( randu() < 0.2 ) ? NaN : ( randu()*100.0 ) - 50.0;
+	v2 = ( randu() < 0.2 ) ? NaN : ( randu()*100.0 ) - 50.0;
+	err = accumulator( v1, v2 );
+	console.log( '%d\t%d\t%d', v1.toFixed( 3 ), v2.toFixed( 3 ), err.toFixed( 3 ) );
```
When `err` is `null` (e.g., first iteration returns NaN), `null.toFixed(3)` throws. Add a null guard. Also, `.toFixed()` returns a string, so use `%s` instead of `%d`.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmae/lib/main.js:
```diff
@@ -0,0 +1,84 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2020 The Stdlib Authors.
```
Copyright year should be 2025, not 2020. Same in all other files in this package.

Planeshifter committed:
fix: address review feedback

Fix double period in package description and guard against null in the
example when computing the mean absolute error over NaN-containing data.

Planeshifter commented:
/stdlib merge

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/nanmae

Planeshifter committed:
chore: update copyright years

Planeshifter committed:
style: use format for benchmark names in stats/incr/nanmae

Planeshifter committed:
docs: remove development artifact comment in stats/incr/nanmae benchmark

Planeshifter committed:
test: use is-nan assertion in stats/incr/nanmae tests

Planeshifter committed:
docs: remove hand-authored See Also entries

The related-packages section is autopopulated by tooling; empty the
`See Also` block and related-links so entries are not authored by hand.

Planeshifter committed:
docs: use backticks around `NaN` in descriptions

Match the sibling convention of formatting `NaN` as inline code in the
main and index JSDoc descriptions.

Planeshifter reviewed (approved):
Let's land this PR!
